infinite-scroll/guide.md

```diff
@@ -0,0 +1,414 @@
+
+Start out with this:
+
+```
+var infiniteScrollWidget = function(options){
+  var container = document.querySelector(options.container);
+  var templates = options.templates;
+
+  if(!container) {
+    throw new Error('infiniteScroll: cannot select \'' + options.container + '\'');
+  }
+
+  return {
+    init: function(){},
+    render: function(args){}
+  }
+};
+```
+
+When creating a custom widget we can expose two or three methods. Both `init` and `render` are always required.
+
+The `init` method initializes the widget before the first search. It accepts an object that has the search `state`, the search `helper`, and the configuration of the templates: the `templatesConfig`.
+
+Meanwhile, the `render` method is called after the search results are returned from Algolia. This also accepts an object with the `results` of the query, the search `state`, the search `helper`, and a function (`createURL`) to create a URL with the search parameters.
+
+Optionally, you can expose a `getConfiguration` method, which can configure the underlying AlgoliaSearch JS helper. It takes [SearchParameters](https://community.algolia.com/algoliasearch-helper-js/docs/SearchParameters.html) and is expected to return an object of configuration properties.
+
+## Rendering Results
+
+To render our results we will leverage [mustache](https://github.com/janl/mustache.js) templates. Place them at the bottom of the index.html:
+
+```
+<script type="text/html" id="hits-template">
+  <div class="hits">
+    {{#pageNo}}
+      <div class="page-no">
+        Page {{pageNo}}
+      </div>
+    {{/pageNo}}
+    {{#hits}}
+      <div class="hit">
+        <a href="{{url}}">
+          <img src="{{image}}" alt="{{name}}">
+          <p class="name">
+            {{#_snippetResult}}
+              {{{_snippetResult.name.value}}}
+            {{/_snippetResult}}
+            {{^_snippetResult}}
+              {{name}}
+            {{/_snippetResult}}
+          </p>
+        </a>
+      </div>
+    {{/hits}}
+  </div>
+</script>
+
+<script type="text/html" id="no-results-template">
+  <div id="no-results-message">
+    <p>We didn&rsquo;t find any results for the search <em>&ldquo;{{query}}&rdquo;</em></p>
+    <a href="#" class="clear-all">Clear all</a>
+  </div>
+</script>
+```
+
+The first template will be filled with information from each search hit, while the second template will be rendered if we have no results.
+
+Let's create a function to render out our templates:
+
+```
+var renderTemplate = function(template, res){
+  var results = document.createElement('div');
+  results.innerHTML = Mustache.render(template, res);
+  return results;
+};
+```
+
+Then we'll create a function to render our search results after a search is made:
+
+```
+var initialRender = function(container, args, templates, parent){
+  if(args.results.nbHits) {
+    _.assign(args.results, {pageNo: page + 1});
```

Comment:
Yeah, you're right, the guide doesn't mention that we've included lodash. I'm not a fan of the imperative style of `args.results.pageNo = page + 1`, but you're right that for a guide we should simplify. (BTW, the vanilla JS equivalent would be `Object.assign(args.results, {pageNo: page + 1}`.

It could be `args.results.page + 1`--there was, at least in the early stages of building this, a need for a shared `page` variable and I didn't want to switch back and forth between that variable and the results, but let me try it again.